pkg_resources/__init__.py

```diff
@@ -363,7 +400,9 @@ class UnknownExtra(ResolutionError):
 DEVELOP_DIST = -1
 
 
-def register_loader_type(loader_type, provider_factory):
+def register_loader_type(
+    loader_type: Type[_ModuleLike], provider_factory: _AdapterType
```

Comment:
See comment https://github.com/pypa/setuptools/pull/4246/files#r1596038268
This would become
```py
def register_loader_type(
    loader_type: Type[_ModuleLike],
    provider_factory: _ProviderFactoryType,
) -> None: ...
```